src/Storages/RocksDB/EmbededRocksDBSettings.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <Core/Defines.h>
+#include <Core/BaseSettings.h>
+#include <Core/SettingsEnums.h>
+#include <Interpreters/Context_fwd.h>
+
+
+namespace Poco::Util
+{
+    class AbstractConfiguration;
+}
+
+
+namespace DB
+{
+class ASTStorage;
+class ASTSetQuery;
+
+#define LIST_OF_ROCKSDB_SETTINGS(M, ALIAS) \
```

Comment:
Why do we need to introduce settings for rocksdb?

Reply:
As far as I understand, enabling the `optimize_trivial_approximate_count_query` setting is for determining whether count approximation can be used while querying `system.tables`.
However, I also doubt introducing settings specific to the RocksDB engine. Why can't we just enable the `optimize_trivial_approximate_count_query` setting globally for the server? I assume users might want it enabled or disabled for all tables using the engine.

Reply:
My initial idea was that by default, this setting should be turned on (at least I do not know why it should be avoided not to use this optimization) also, `total_rows` and `total_bytes` should be in the system tables independently of the `optimize_trivial_approximate_count_query` setting.

> Why do we need to introduce settings for rocksdb?

I like how settings related only to one engine are placed in their own settings file.

> However, I also doubt introducing settings specific to the RocksDB engine. Why can't we just enable the optimize_trivial_approximate_count_query setting globally for the server? I assume users might want it enabled or disabled for all tables using the engine.

As I mentioned in the initial idea, I am reckoning that this setting should be turned on by default, but for some reason, a user can disable it. I thought it was easier to do it with a table setting, but if you are counting that a global setting works better, no objections. 

But I agree, now it's unnecessary complexity, I'll rewrite it.

Reply:
So, I rethought the changes, tried to simplify them, and avoid incompatibilities. 

 Also added a short description in the documentation about `optimize_trivial_approximate_count_query`.

In short, everything works as before except now there is a total_bytes in `system.tables`, total_rows can be shown with optimize_trivial_approximate_count_query setting.